protocols/kad/src/query.rs

```diff
@@ -84,42 +88,42 @@ impl<TInner> QueryPool<TInner> {
     }
 
     /// Returns an iterator that allows modifying each query in the pool.
-    pub(crate) fn iter_mut(&mut self) -> impl Iterator<Item = &mut Query<TInner>> {
+    pub(crate) fn iter_mut(&mut self) -> impl Iterator<Item = &mut Query> {
         self.queries.values_mut()
     }
 
     /// Adds a query to the pool that contacts a fixed set of peers.
-    pub(crate) fn add_fixed<I>(&mut self, peers: I, inner: TInner) -> QueryId
+    pub(crate) fn add_fixed<I>(&mut self, peers: I, info: QueryInfo) -> QueryId
     where
         I: IntoIterator<Item = PeerId>,
     {
         let id = self.next_query_id();
-        self.continue_fixed(id, peers, inner);
+        self.continue_fixed(id, peers, info);
         id
     }
 
     /// Continues an earlier query with a fixed set of peers, reusing
     /// the given query ID, which must be from a query that finished
     /// earlier.
-    pub(crate) fn continue_fixed<I>(&mut self, id: QueryId, peers: I, inner: TInner)
+    pub(crate) fn continue_fixed<I>(&mut self, id: QueryId, peers: I, info: QueryInfo)
     where
         I: IntoIterator<Item = PeerId>,
     {
         assert!(!self.queries.contains_key(&id));
         let parallelism = self.config.replication_factor;
         let peer_iter = QueryPeerIter::Fixed(FixedPeersIter::new(peers, parallelism));
-        let query = Query::new(id, peer_iter, inner);
+        let query = Query::new(id, peer_iter, info);
         self.queries.insert(id, query);
     }
 
     /// Adds a query to the pool that iterates towards the closest peers to the target.
-    pub(crate) fn add_iter_closest<T, I>(&mut self, target: T, peers: I, inner: TInner) -> QueryId
```

Comment:
Remark: It uses to be possible to add initial addresses through `inner.addresses`, it isn't possible anymore.

I couldn't find anywhere where a non-empty `inner.addresses` was provided though, so it is probably fine to keep it like this.

Reply:
yeah YAGNI :) if we need it back we'll add it again